File: day20/src/main/java/de/havox_design/aoc2016/day20/Day20.java
package de.havox_design.aoc2016.day20;

import de.havox_design.utils.DataReader;

import java.util.List;

public class Day20 {
    private final List<String> input;

    public Day20(String fileName) {
        input = readData(fileName);
    }

    public static long solvePart1(String fileName) {
        Day20 instance = new Day20(fileName);
        return instance.solvePart1();
    }

    public static long solvePart2(String fileName) {
        Day20 instance = new Day20(fileName);
        return instance.solvePart2();
    }

    public long solvePart1() {
        return 0L;
    }

    public long solvePart2() {return 0L;
    }

    private List<String> readData(String fileName) {
        return DataReader.readData(fileName, MainClass.class);
    }
}

File: templates/scala/aoc20xx-scala/src/main/scala/de/havox_design/aoc20xx/day20/Day20.scala
package de.havox_design.aoc2022.day20

object Day20 {
  def solvePart1(filename: String): Long = {
    1L
  }

  def solvePart2(filename: String): Long = {
    1L
  }

  def main(args: Array[String]): Unit = {
    println("Solution for part1: " + solvePart1("de/havox_design/aoc20xx/day20/day20.txt"))
    println("Solution for part2: " + solvePart2("de/havox_design/aoc20xx/day20/day20.txt"))
  }

  private def readData(filename: String): Iterator[String] =
    scala
      .io
      .Source
      .fromResource(filename)
      .getLines()
}
